velox/functions/sparksql/tests/ArithmeticTest.cpp

```diff
@@ -761,6 +761,37 @@ TEST_F(ArithmeticTest, abs) {
       std::numeric_limits<double>::max());
 }
 
+TEST_F(ArithmeticTest, absAnsiMode) {
+  // Enable ANSI mode
+  queryCtx_->testingOverrideConfigUnsafe(
+      {{core::QueryConfig::kSparkAnsiEnabled, "true"}});
+
+  // Normal cases should still work
+  EXPECT_EQ(abs<int8_t>(-127), 127);
+  EXPECT_EQ(abs<int16_t>(-32767), 32767);
+  EXPECT_EQ(abs<int32_t>(-2147483647), 2147483647);
+  EXPECT_EQ(abs<int64_t>(-9223372036854775807), 9223372036854775807);
+
+  // Minimum values should throw in ANSI mode
+  EXPECT_THROW(abs<int8_t>(std::numeric_limits<int8_t>::min()), VeloxUserError);
+  EXPECT_THROW(
+      abs<int16_t>(std::numeric_limits<int16_t>::min()), VeloxUserError);
+  EXPECT_THROW(
+      abs<int32_t>(std::numeric_limits<int32_t>::min()), VeloxUserError);
+  EXPECT_THROW(
+      abs<int64_t>(std::numeric_limits<int64_t>::min()), VeloxUserError);
+
+  // Floating point should still work normally (no overflow for floats)
+  EXPECT_EQ(abs<float>(-99999.9999f), 99999.9999f);
+  EXPECT_EQ(
+      abs<float>(std::numeric_limits<float>::lowest()),
+      std::numeric_limits<float>::max());
+  EXPECT_EQ(abs<double>(-99999.9999), 99999.9999);
+  EXPECT_EQ(
+      abs<double>(std::numeric_limits<double>::lowest()),
+      std::numeric_limits<double>::max());
+}
```

Comment:
Do we still need to test normal cases? It seems repetitive. Perhaps we could focus on testing only the divergent cases.

Reply:
I reviewed Spark's test code in the link below. For common cases, they test both ANSI on and off within a for loop, expecting identical behavior. For divergent cases, they use different expectations. This approach avoids duplicate test code. Could we adopt this methodology in Velox to test all ANSI-related functions?

https://github.com/apache/spark/blob/5a30d68a652075497d72985fffb31ee281b18c79/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ArithmeticExpressionSuite.scala#L512-L556